dms_version/models/dms_storage.py

```diff
@@ -0,0 +1,65 @@
+# Copyright 2017-2020 MuK IT GmbH
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class DmsStorage(models.Model):
+    _inherit = "dms.storage"
+
+    has_versioning = fields.Boolean(
+        string="Active Versioning",
+        default=False,
+        help="Indicates if files have an active version control.",
+    )
+    stored_versions = fields.Integer(
+        string="Saved Versions",
+        default=3,
+        help="Indicates how many versions are saved per file.",
+    )
+    clean_versions = fields.Selection(
+        selection=[("immediately", _("On Write")), ("autovacuum", _("On Autovacuum"))],
```

Comment:
Never use translations outside a context where there's a `self.env` available. Otherwise there's no place where to get `self.env.lang` from, and the translations do nothing.

OTOH, "On Autovacuum" is a bit too technical for users:

```suggestion
        selection=[("immediately", "Immediately"), ("autovacuum", "Scheduled")],
```

Finally... do we really need something like this? 🤔 Why should a user care about when versions are cleared? Can't we just do it on autovacuum always and add a note in the form explaining why they won't disappear immediately?

If that's not possible, wouldn't it be better a bool field? `clean_old_versions_immediately` -> yes/no